web-app-router/scenes/Portal/Teams/TeamId/Apps/AppId/Actions/ActionId/Settings/UpdateAction/index.tsx

```diff
@@ -0,0 +1,164 @@
+
+  const submit = useCallback(
+    async (values: NewActionFormValues) => {
+      try {
+        const result = await updateActionQuery({
+          variables: {
+            id: action.id,
+            input: {
+              name: values.name,
+              description: values.description,
+              max_verifications: values.maxVerifications,
+            },
+          },
+        });
+
+        if (result instanceof Error) {
+          throw result;
+        }
+        // TODO: Turn on Posthog
+        // posthog.capture("action_created", {
+        //   name: values.name,
+        //   app_id: currentApp.id,
+        //   action_id: values.action,
+        // });
+      } catch (error) {
+        console.error(error);
+        return toast.error("Error occurred while updating action.");
+      }
+      toast.success(`Action "${values.name}" updated.`);
+    },
+    [updateActionQuery, action.id],
+  );
+
+  const copyAction = useCallback(() => {
+    navigator.clipboard.writeText(watch("action"));
+    toast.success("Copied to clipboard");
+  }, [watch]);
+
+  return (
+    <div className="w-full">
+      <div className="grid items-center justify-center min-h-full">
+        <form
+          onSubmit={handleSubmit(submit)}
+          className="grid grid-cols-1 gap-5"
+        >
+          <h1 className="text-lg font-[550] mb-2 text-grey-900">Settings</h1>
+          <Input
+            register={register("name")}
+            errors={errors.name}
+            label="Action Name"
+            placeholder="Anonymous Vote #12"
+            required
+            className="w-136"
+          />
+          <Input
+            register={register("description")}
+            errors={errors.description}
+            label="Short Description"
+            placeholder="Cast your vote on proposal #102"
+            helperText="Tell your users what the action is about. Shown in the World App."
+            required
+            className="w-136"
+          />
+          <Input
+            register={register("action")}
+            errors={errors.action}
+            label="Identifier"
+            helperText="This is the value you will use in IDKit and any API calls."
+            placeholder="A short description of your action"
+            disabled
+            addOnPosition="right"
+            addOn={
+              <button className="px-1" type="button" onClick={copyAction}>
+                <CopyIcon />
+              </button>
+            }
+            className="w-136 text-grey-400"
+          />
+          <Controller
+            name="maxVerifications"
+            control={control}
+            render={({ field }) => {
+              return (
+                <MaxVerificationsSelector
+                  value={field.value}
+                  onChange={field.onChange}
+                  errors={errors.maxVerifications}
+                  showCustomInput
+                  className="w-136" // border is 2 px
```

Comment:
@andy-t-wang Why do we need to hardcode input's height? let's make it to just use parent's width `w-full`